### PR TITLE
Convert Emacs documentation from XML DocBook to Commonmark

### PR DESCRIPTION
**Pull request by AndersonTorres**
https://github.com/NixOS/nixpkgs/issues/105254

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: doc/builders/packages/index.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
  <xi:include href="dlib.xml" />
  <xi:include href="eclipse.xml" />
  <xi:include href="elm.xml" />
- <xi:include href="emacs.xml" />
+ <xi:include href="emacs.section..xml" />
```
Comment (taku0) [inline doc/builders/packages/index.xml:12]:
```suggestion
 <xi:include href="emacs.section.xml" />
```

**Comment by AndersonTorres** [comment]
@ryantm ping!

File: doc/builders/packages/emacs.section.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    ]));
+  };
+}
+```
+
+This provides a fairly full Emacs start file. It will load in addition to the user's presonal config. You can always disable it by passing `-q` to the Emacs command.
+
+Sometimes `emacsWithPackages` is not enough, as this package set has some priorities imposed on packages (with the lowest priority assigned to Melpa Unstable, and the highest for packages manually defined in `pkgs/top-level/emacs-packages.nix`). But you can't control this priorities when some package is installed as a dependency. You can override it on per-package-basis, providing all the required dependencies manually - but it's tedious and there is always a possibility that an unwanted dependency will sneak in through some other package. To completely override such a package you can use `overrideScope'`.
+
+```
```
Comment (ryantm) [inline doc/builders/packages/emacs.section.md:108]:
```suggestion
```nix
```

**Review by ryantm — CHANGES_REQUESTED** [review]

File: doc/builders/packages/emacs.section.md (hunk 2 above)
Comment (AndersonTorres) [reply doc/builders/packages/emacs.section.md]: done

**Review by ryantm — APPROVED** [review]

**Comment by ryantm** [comment]
@AndersonTorres looks good thanks!

File: doc/builders/packages/emacs.section.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
+# Emacs {#sec-emacs}
+
+## Configuring Emacs
```
Comment (jtojnar) [inline doc/builders/packages/emacs.section.md:3]: Looks like anchor is missing here.

Comment (ryantm) [reply doc/builders/packages/emacs.section.md:3]: Thanks for fixing it!

**Pull request merged**